### ISSUE TITLE
Status on this library?

### ISSUE THREAD

**Issue by AndrewSB** [drumnkyle/music-notation-swift#17]
Hey!

I'm going to be working on a music learning iOS app starting next week and I'd love to use your library.
I know it's a WIP, how close is it to a buildable beta?

**Comment by AndrewSB** [comment]
Is there anything I can contribute that would get us to that point otherwise? My alternative would be to probably reimplement this library myself, so I'd prefer to work on this together 😄

**Comment by drumnkyle** [comment]
Hi Andrew. Thanks so much for your interest. Myself and @migue48 have gotten more serious about this recently and are getting closer to completing the base functionality. It depends on what exactly you expect from it. It's mostly me doing the actual coding since @migue48 is brand new to Swift. We also both have full time jobs so this is done in our spare time. 

Any help would be greatly appreciated. I see it looks like you are located in the San Francisco area. I work in SF. Let me know if you'd be interested in meeting in person so we can talk about details. 

What platform(s) are you planning on making your app for?

**Comment by AndrewSB** [comment]
I'm going to be working on a (iOS) side project that renders sheet music written in http://www.vexflow.com/vextab/, which uses the mic to recognize where you are in the music and tell you if the notes you are playing are correct.

So what I really need is a UI library that can render sheet music on iOS natively - I didn't read over your project closely enough before creating this issue - that's not something you're interested in creating, correct? You're simply creating a new format, not unlike http://www.vexflow.com/vextab/?

**Comment by drumnkyle** [comment]
Ah. Yes, we are not looking to create a UI as part of this library. The idea of this library is to essentially be the data layer. A UI can then be built on top of this in either a cross-platform or platform-specific way. I hadn't heard of VexTab, so thanks for letting me know about that. Seems interesting. I'll have to look into it more.

Let me know if you are not interested in this library then, and I will close this issue.

**Comment by AndrewSB** [comment]
Yep, Vextab already solves the problem (for me) that this library is working on.

Good luck! Do let me know if you guys are thinking about rendering data as tablature, I'd be interested in that!

**Comment by AndrewSB** [comment]
@drumnkyle would you be interested in collaborating on a sheet music rendering library?

**Comment by drumnkyle** [comment]
Open source? Yes, we were planning to do that after we completed this. However, I would be looking to make it work with this library. If you'd like to meet, we could talk more about what you're looking for and your timelines. At the least, I would be more than happy lend support and help with design and architecture.

**Comment by davidfloegel** [comment]
Hey guys,

I've been watching this project for a while as I'm planning an iOs app that definitely needs music notation rendering.

Would this library be sufficient enough already to display simple chords, scales and intervals and maybe even rhythm notations (slash notation for example?)

Thanks!

**Comment by drumnkyle** [comment]
Hi @davidfloegel. The library is definitely not in a fully working state yet. As far as rendering actual sheet music, this library on its own will not do that. This library only holds the notation and does validation of timing and things like that. Basically it's the data layer. We plan to make a separate library on top of this for rendering the notation. 

Let me know if you are interested in helping out. The faster we get the data later done, the faster we can get to the rendering portion. ;-)

Also, what do you mean by slash notation?

**Comment by davidfloegel** [comment]
Hey man,

aaah ok. Do you know of any swift library that can render music notations? As I couldn't find one by just googling.
I would love to help out but I think my swift knowledge is not good enough yet haha!

And slash notation is a style to notate rhythms. 
https://musescore.org/sites/musescore.org/files/Rhythmic%20slash_notation.png

Cheers
David

**Comment by drumnkyle** [comment]
I didn't see any either. That's why I started creating this. We are moving along but still have a while before we get to the actual rendering.

**Comment by davidfloegel** [comment]
Do you maybe want to ping me an email or contact me on skype and tell me what's left to do and maybe I can actually help? :) Would be very interesting!

**Comment by drumnkyle** [comment]
@davidfloegel I tried emailing you but you haven't responded. Let me know if you are still interested at all. I have listed everything that is left to do in the issues. However, the library should be functional before all of them are completed.
